clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -3831,7 +3845,12 @@ private void testOffsetValidationWithGivenEpochOffset(final EpochEndOffset epoch
         consumerClient.poll(time.timer(Duration.ZERO));
 
         assertEquals(0, subscriptions.position(tp0).offset);
-        assertFalse(subscriptions.awaitingValidation(tp0));
+
+        if (offsetResetStrategy == OffsetResetStrategy.NONE) {
+            assertTrue(subscriptions.awaitingValidation(tp0));
+        } else {
+            assertFalse(subscriptions.awaitingValidation(tp0));
```

Comment:
Hmm.. Can we assert the raised exception somehow? It's not clear to me that it is getting raised appropriately and we don't have any tests for it.

Reply:
As discussed offline, we plan to address this gap in a separate PR for https://issues.apache.org/jira/browse/KAFKA-10087 as some other bug to fix for log truncation exception propagation.